src/librustdoc/html/render/print_item.rs

```diff
@@ -1509,7 +1514,9 @@ fn render_struct(
             if let Some(g) = g {
                 write!(w, "{}", print_where_clause(g, cx, 0, false),)
             }
-            w.write_str(";");
+            if structhead {
+                w.write_str(";");
+            }
```

Comment:
Otherwise, for the tuple struct enum variant, we get:

```rust
enum Foo {
    Bar(u32);,
}
```

Which seems not what we want. I'll add a comment though.